docs/visualization/histogram.rst

```diff
@@ -0,0 +1,118 @@
+
+Astropy's :func:`~astropy.visualization.hist` function addresses this by providing several methods of automatically tuning the histogram bin size. It has a syntax identical to matplotlib's ``plt.hist`` function, with the exception of the ``bins`` parameter, which allows specification of one of four different methods for automatic bin selection. These methods are implemented in :func:`astropy.stats.histogram`, which has a similar syntax to the ``np.histogram`` function.
+
+Normal Reference Rules
+----------------------
+The simplest methods of tuning the number of bins are the normal reference rules due to Scott (implemented in :func:`~astropy.stats.scott_bin_width`) and Freedman & Diaconis (implemented in :func:`~astropy.stats.freedman_bin_width`).
+These rules proceed by assuming the data is close to normally-distributed, and applying a rule-of-thumb intended to minimize the difference between the histogram and the underlying distribution.
+
+The following figure shows the results of these two rules on the above dataset:
+
+.. plot::
+   :align: center
+
+    import numpy as np
+    from astropy.visualization import hist
+
+    # generate some complicated data
+    rng = np.random.RandomState(0)
+    t = np.concatenate([-5 + 1.8 * rng.standard_cauchy(500),
+                        -4 + 0.8 * rng.standard_cauchy(2000),
+                        -1 + 0.3 * rng.standard_cauchy(500),
+                        2 + 0.8 * rng.standard_cauchy(1000),
+                        4 + 1.5 * rng.standard_cauchy(1000)])
+    
+    # truncate to a reasonable range
+    t = t[(t > -15) & (t < 15)]
+
+    # draw histograms with two different bin widths
+    fig, ax = plt.subplots(1, 2, figsize=(10, 4))
+    hist_kwds1 = dict(histtype='stepfilled', alpha=0.2, normed=True)
+
+    fig.subplots_adjust(left=0.1, right=0.95, bottom=0.15)
+    for i, bins in enumerate(['scott', 'freedman']):
+        hist(t, bins=bins, ax=ax[i], histtype='stepfilled', alpha=0.2, normed=True)
+        ax[i].set_xlabel('t')
+        ax[i].set_ylabel('P(t)')
+        ax[i].set_title('hist(t, bins="{0}")'.format(bins),
+                        fontdict=dict(family='monospace'))
+
+
+As we can see, both of these rules of thumb choose an intermediate number of bins which provide a good tradeoff between data representation and noise suppression.
+
+Bayesian Models
+---------------
+
+Though rules-of-thumb like Scott's rule and the Freedman-Diaconis rule are fast and convenient, their strong assumptions about the data make them suboptimal for more complicated distributions. Other methods of bin selection use fitness functions computed on the actual data to choose an optimal binning. Astropy implements two of these examples: Knuth's rule (implemented in :func:`~astropy.stats.knuth_bin_width`) and Bayesian Blocks (implemented in :func:`~astropy.stats.bayesian_blocks`).
```

Comment:
Could you please hard-wrap all text to ~ 80 characters?
(If I'm not mistaken we do this everywhere in Astropy and it helps with git diffs)